imap_processing/codice/tests/test_decom.py

```diff
@@ -0,0 +1,104 @@
+"""Tests the decommutation process for CoDICE CCSDS Packets"""
+
+from pathlib import Path
+
+import pandas as pd
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+
+
+@pytest.fixture(scope="session")
+def decom_test_data():
+    """Read test data from file"""
+    packet_file = Path(
+        f"{imap_module_directory}/codice/tests/data/"
+        f"raw_ccsds_20230822_122700Z_idle.bin"
+    )
+    xtce_document = Path(
+        f"{imap_module_directory}/codice/packet_definitions/P_COD_NHK.xml"
+    )
+    data_packet_list = decom_packets(packet_file, xtce_document)
+    data_packet_list = [
+        packet
+        for packet in data_packet_list
+        if packet.header["PKT_APID"].raw_value == 1136
+    ]
+
+    return data_packet_list
+
+
+@pytest.fixture(scope="session")
+def validation_data():
+    """Read in validation data from the CSV file
+
+    Returns
+    -------
+    validation_data : pandas DataFrame
+        The validation data read from the CSV, cleaned up and ready to compare
+        the decommutated packet with
+    """
+
+    # Read in the CSV file (perhaps to a pandas dataframe?)
+    validation_file = Path(
+        f"{imap_module_directory}/codice/tests/data/"
+        f"idle_export_raw.COD_NHK_20230822_122700.csv"
+    )
+    validation_data = pd.read_csv(validation_file)
+    # Remove the timestamp column and data
+    if "timestamp" in validation_data.columns:
+        validation_data = validation_data.drop(columns=["timestamp"])
+
+    # Return the data
+    return validation_data
+
+
+def test_housekeeping_data(decom_test_data, validation_data):
+    """Compare the decommutated housekeeping data to the validation data.
+
+    Parameters
+    ----------
+    decom_test_data : List
+        The decommuted housekeeping packet data
+    validation_data : pandas DataFrame
+        The validation data to compare against
+    """
+
+    # Take the first decom_packet
+    first_data = decom_test_data[0]
+
+    # Get the corresponding row in validation_data based on the "SHCOARSE" value
+    validation_row = validation_data.loc[first_data.data["SHCOARSE"].raw_value]
+
+    # Compare raw values of housekeeping data
+    for key, value in first_data.data.items():
+        if key == "SHCOARSE":
+            # Compare SHCOARSE value
+            assert value.raw_value == validation_row.name
+            continue
+
+        # Check if the key is present in validation_row
+        if key not in validation_row.index:
+            print(f"Key '{key}' not found in validation_row. Skipping comparison.")
+            continue
```

Comment:
So this is suggesting that there are key/value pairs in the housekeeping packet that are not in the validation file? Would this then mean that the validation data are missing some items? 

If that is the case, this is probably something we want to ask Joey and/or Greg Dunn about. I would think we would ideally want a one-to-one mapping?

Reply:
I agree with others. If key is not in validation data, then packet definitions may have changed. It should fail if it didn't find the key.